app/src/main/java/com/marvel/marvel/main/model/QueryMap.kt

```diff
@@ -0,0 +1,56 @@
+package com.marvel.marvel.main.model
+
+
+import android.util.Log
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.*
+
+internal class QueryMap : LinkedHashMap<String, String>() {
+
+    companion object {
+        private const val PUBLIC_KEY = "c735e50e07fb1d2384b908dea2ef423a" //TODO inject and move to environment variable
+        private const val PRIVATE_KEY = "139ad61ebc977dc1dbdfb84f26f11d6f9985dd29"//TODO inject and move to environment variable
+        private const val DATE_DESCRIPTOR_KEY = "dateDescriptor"
+        private const val DATE_DESCRIPTOR_VALUE = "thisMonth"
+        private val TAG = QueryMap::class.simpleName
+        private const val LIMIT_KEY = "limit"
+        private const val LIMIT = "100"
+        private const val TIMESTAMP_KEY = "ts"
+        private const val API_KEY = "apikey"
+        private const val HASH_KEY = "hash"
+    }
+
+    init {
+        val timestamp = System.currentTimeMillis().toString()
+        val hash = md5(timestamp + PRIVATE_KEY + PUBLIC_KEY)
+        put(LIMIT_KEY, LIMIT)
+        put(TIMESTAMP_KEY, timestamp)
+        put(API_KEY, PUBLIC_KEY)
+        put(HASH_KEY, hash)
+        put(DATE_DESCRIPTOR_KEY, DATE_DESCRIPTOR_VALUE)
+    }
+
+    private fun md5(string: String): String {
+        val md5 = "md5"
+        try {
+            // Create md5 Hash
+            val digest = MessageDigest.getInstance(md5)
+            digest.update(string.toByteArray())
+            val messageDigest = digest.digest()
+            // Create Hex String
```

Comment:
Comments are a sign that your code is [doing too much](https://en.wikipedia.org/wiki/Single_responsibility_principle). 

If this were a nice load of Kotlin extensions methods this could be:

```
private fun String.md5String() = md5Binary().toHexString()

private fun String.md5Binary(): ByteArray = ...

private fun ByteArray.toHexString() = ...
```

There's no need for comments because the names of the functions document all you need to know.